### PR TITLE
fix `gasPrice`, `effectiveGasPrice`, and `gasUsed`

### PR DESCRIPTION
**Pull request by shunjizhan**
## Change
use the latest formula to calculate correct tx fee, so then can get correct gas info
```
ActualFee = BaseFee + LenFee + (adjustedWeightFee / EstimateWeight * ActualWeight)
```

also had some linter fix, now `rush build` can finished without any warning

## Test
- for [this tx](https://blockscout.mandala.acala.network/tx/0xf2dc76aa6cbaaf4ddd11ec26211a558f88785fb149aff0a2ee1785d5be3dd165/internal-transactions) previously `gasPrice * gasUsed !== actualTxFee (3.323382059983 )`, now it is correct
- added new tests to endpoint tests to check the calculatedGas is 99.9999% equal actual tx fee (due the rounding issue it usually can't be exactly the same).

### CODE REVIEW

File: eth-providers/src/base-provider.ts [hunk 1]
```diff
@@ -706,7 +714,7 @@ export abstract class BaseProvider extends AbstractProvider {
     const blockTag = await this._ensureSafeModeBlockTagFinalization(_blockTag);
 
     // @TODO resolvedPosition
-    const { address, blockHash, resolvedPosition } = await resolveProperties({
+    const { address, blockHash } = await resolveProperties({
       address: this._getAddress(addressOrName),
       blockHash: this._getBlockHash(blockTag),
       resolvedPosition: Promise.resolve(position).then((p) => hexValue(p))
```
Comment (zjb0807) [inline eth-providers/src/base-provider.ts:720]: why not remove this line?

Comment (shunjizhan) [reply eth-providers/src/base-provider.ts]: I am actually not sure how `resolvedPosition` will be used for the @todo, let me just add it back and ignore the build warning

Comment (shunjizhan) [reply eth-providers/src/base-provider.ts]: @ntduan is the `resolvedPosition` still useful in the future? If not I can remove it

**Comment by shunjizhan** [comment]
I am gonna merge this and if any other small changes need I can do it in a separate PR

**Comment by shunjizhan** [comment]
CI should pass just oddly getting timeout #475

**Comment by shunjizhan** [comment]
@zjb0807 let me know if you need help resolving any merge conflict, since there are many structural change recently

**Pull request merged**